core/src/main/scala/nl/vroste/zio/kinesis/client/zionative/LockingCheckpointer.scala

```diff
@@ -0,0 +1,107 @@
+package nl.vroste.zio.kinesis.client.zionative
+
+import nl.vroste.zio.kinesis.client.Record
+import zio._
+
+final class LockingCheckpointer[T] private (
```

Comment:
Can we think of a way to offer this functionality when users are using `shardedStream`?

Reply:
I think you would have to manually wire it, same as with the regular checkpointing behavior. That one also doesn't work with shardedStream out of the box if I remember correctly.